### DISCUSSION TITLE
Search for #Hashtag in Messages

### DISCUSSION THREAD

**Discussion by mkarg** [element-hq/element-meta#562] [Ideas]
#### What would you like to do?

I want to add tags to messages and search all messages by tag.

#### Why would you like to do it?

Because I am used to do that from social media.

Because it is convenient to search all messages ever sent and received to easily find everthing around a shared topic, independent of the chat room and chat topic.

#### How would you like to achieve it?

I want to simply write `#mytopic` within a text message. This will add the tag "mytopic" to this message. Element shall render `#mytopic` as a clickable link. When I click it, Element shall list all messages found containing the same hashtag.

### Have you considered any alternatives?

An alternative would be the ability to explicitly add labels to messages, which is not as typical in social media and rather cumbersome.

**Comment by adriel-tech** [comment]
I add #hashtags to messages I have sent my self, then search for the hashtag, works fine. I understand you are looking for something a little more specific, but this can be currently done, you just have to search the tag not just click a link. Not saying this is not a good suggestion.

**Comment by mkarg** [comment]
Well, as an end user, I *want* to just click the link, as searching is *PITA* and holding back people from switching to Matrix. So what now?
